challenges-14-plus.yml

```
@@ -0,0 +1,472 @@
+      choices:
+        a: |
+          ```javascript
+          document.addEventListener('turbo:visit', () => {
+              document.body.classList.add('turbo-loading');
+          });
+          ```
+        b: |
+          ```javascript
+          document.addEventListener('turbo:visit', () => {
+              document.body.classList.add('turbo-loading');
+          });
+          
+          document.addEventListener('turbo:before-render', (event) => {
+              event.detail.newBody.classList.add('turbo-loading');
+          });
+          
+          document.addEventListener('turbo:render', () => {
+              requestAnimationFrame(() => {
+                 document.body.classList.remove('turbo-loading');
+              });
+          });
+          ```
+        c: |
+          ```javascript
+          document.addEventListener('turbo:visit', () => {
+              document.body.classList.add('turbo-loading');
+          });
+
+          document.addEventListener('turbo:before-render', (event) => {
+              event.detail.newBody.classList.add('turbo-loading');
+          });
+          ```
+        d: |
+          ```javascript
+          document.addEventListener('turbo:load', () => {
+              document.body.classList.add('turbo-loading');
+          });
+
+          document.addEventListener('turbo:before-render', (event) => {
+              event.detail.newBody.classList.add('turbo-loading');
+          });
+
+          document.addEventListener('turbo:render', () => {
+              requestAnimationFrame(() => {
+                 document.body.classList.remove('turbo-loading');
+              });
+          });
+          ```
+      correct: b
+  
+transitions_2:
+  question: |
+    Why is it hard in Turbo to implement page load transitions in versions 7.0.0 beta?
+    
+    > Notice that starting with version 7.0.0-rc.1 the "Pausable rendering" PR has been
+    > merged, making it easier to implement page transitions between clicks!
+  explanation: |
+    The fact that we cannot pause a render operation in Turbo before 7.0.0-rc.1 means that
+    our transition, if we make one, will always take the time it takes for the Ajax call
+    to finish. This makes it hard for us to perform complex transitions when we can't
+    *pause* the render until our animation properly finishes.
+  features:
+    multiple_chopice:
+      a: |
+        Because there are no good event handlers to use when transitioning pages for
+        this particular purpose.
+      b: |
+        Actually, you *can* easily implement page transitions between clicks in 7.0.0 beta.
+      c: |
+        Because wnen a page finishes loading, Turbo replaces the `<body>` element
+        immediately, leaving no time for a transition.
+      d: |
+        Because there are other libraries that can achieve this.
+    correct: c
+
+transition_finish:
+  question: |
+    Our lead designer is *ANGRY*! When our page went live, we've discovered that some
```

```suggestion
    Our lead designer is *annoyed*! When our page went live, we've discovered that some
```

We all can use less anger 😆